core_engine/find_all_url_data.py

```diff
@@ -0,0 +1,169 @@
+import sys
+from bs4 import BeautifulSoup
+import requests
+from text_finder import read_url
+import time
+import sys
+
+sys.setrecursionlimit(100000)
+
+
+def get_all_data_from_url(url, links=[], count=0):
+    try:
+        """Handle multiple repeating of exception"""
+        print("Link...", end=' ')
+        if count > 20:
+            return 0
+        """Reading html page and extract tags with texts"""
+        response = requests.get(url).text
+        soup = BeautifulSoup(response, 'html.parser')
+        th_tags = soup.find_all('th')
+
+        """Check whether list of texts is empty and handle error 503"""
+        if not th_tags:
+            print('\033[1;31;40m Empty list. Wait a bit... \033[0m', end=' ')
+            time.sleep(2)
+            count += 1
+            return get_all_data_from_url(url, links, count)
+
+        """Search for text and add it into a file"""
+        with open("training_data.txt", 'ab') as file:
+            print('Read text', end=' ')
+            file.write(read_url(soup).encode('cp1251', errors='ignore'))
+            print("\033[1;33;40m Text was added \033[0m  ", end=' ')
+
+        find_new_url(th_tags, links)
+
+    except UnicodeEncodeError as e:
+        print("Exception {} is occurred!".format(e))
+        print(th_tags[-1])
+        find_new_url(th_tags, links)
+    except Exception as e:
+        print("Exception \'{0}\' at line {1} is occurred!".format(e, sys.exc_info()[-1].tb_lineno))
+    finally:
+        return 0
+
+
+def find_new_url(th_tags, links):
+        """Search for tag that points to next page. If absent - break since it was the last one."""
+        if str(th_tags[-1]['class'])[2:-2] == 'pagenav_next':
+            a = th_tags[-1].find('a')
+        else:
+            print("Book is finished")
+            time.sleep(5)
+            return 0
+
+        """Make a list with urls"""
+        new_url = a.get('href')
+        if new_url not in links:
+            links.append(new_url)
+            print('New link', end=' ')
+
+        """Add every 10 links into a file"""
+        if len(links) % 10 == 0:
+            with open("links.txt", 'a') as file:
+                file.write('\n'.join(links[-1:-11:-1]))
+                file.write('\n')
+                print('\n', '-'*150, '\nIt was added {} links!\n'.format(len(links)), '-'*150, '\n')
+        print("\033[1;34;40m Finish \033[0m  \n ", '-' * 150)
+
+        get_all_data_from_url(new_url, links)
+
+
+print('Loading', end='')
+books = 0
+start = time.time()
+blogUrl = 'https://pidruchniki.com/81900/menedzhment/upravlinnya_lyudskimi_resursami'
```

Comment:
instead of copy paste it is much better to store urls in list

Reply:
That’s right)
It was just an impulsive decision after program finally started to work and I didn't brake it to set it right.
Just forgot to fix it befor pushing.